File: strategy-game-backend/src/main/kotlin/de/ruegnerlukas/strategygame/backend/ports/provided/game/GameCreateAction.kt
package de.ruegnerlukas.strategygame.backend.ports.provided.game

import de.ruegnerlukas.strategygame.backend.ports.models.world.WorldSettings

/**
 * Create a new game
 */
interface GameCreateAction {

	/**
	 * @return the id of the game
	 */
	suspend fun perform(worldSettings: WorldSettings): String

}

File: strategy-game-frontend/src/core/gameCreateAction.ts
import {GameApi} from "./required/gameApi";

/**
 * Create a new game
 */
export class GameCreateAction {

    private readonly gameApi: GameApi;

    constructor(gameApi: GameApi) {
        this.gameApi = gameApi;
    }

    perform(seed: string | null): Promise<string> {
        console.log("create new game (seed=" + seed + ")");
        return this.gameApi.create(seed);
    }

}